### PR TITLE
Fix iOS native fullscreen captions position

### PR DESCRIPTION
**Pull request by pajong**
### This PR will...
- Remove max-height to the native captions on iOS native fullscreen.
- Remove `max-height` on default captions, and apply it when the controls are showing
- Remove css functions that are not being used

### Why is this Pull Request needed?
- The styles added to the native captions container positions the captions too high on iPhone fullscreen.
- [Tech debt:](https://github.com/jwplayer/jwplayer/issues/2005) we should not put `max-height` by default and then removing that when the controlbar is active. Instead, we should only apply when the controlbar is active.

### Are there any points in the code the reviewer needs to double check?
- This PR is adding a new class `jw-flag-ios-fullscreen`: please comment if there is a better way or better suggestions

#### Addresses Issue(s):

JW8-2660

### CODE REVIEW

**Comment by johnBartos** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  🛠 There are modified src files, but no test changes. Add tests if you're able to.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/js/view/view.js [hunk 1]
```diff
@@ -558,6 +558,7 @@ function View(_api, _model) {
             const provider = model.getVideo() || instreamProvider;
             if (provider && provider.setFullscreen) {
                 provider.setFullscreen(state);
+                toggleClass(_playerElement, 'jw-flag-ios-fullscreen', state);
```
Comment (robwalch) [inline src/js/view/view.js]: This class should be added in response to the native fullscreen event, not following the request to go fullscreen.

File: src/js/view/view.js [hunk 2]
```diff
@@ -636,6 +636,10 @@ function View(_api, _model) {
         return provider.getFullScreen();
     }
 
+    function _nativeFullscreenChangeHandler(event) {
+        toggleClass(_playerElement, 'jw-flag-ios-fullscreen', event.jwstate);
```
Comment (pajong) [inline src/js/view/view.js]: Do we need to do something like this?
`const newState = (event.jwstate !== undefined) ? event.jwstate : _isNativeFullscreen();`
I see that the event is coming from the provider and it is always adding `jwstate`, but the `_fullscreenChangeHandler` below gets the newState like that. Am I missing something?

Comment (robwalch) [reply src/js/view/view.js]: You should only add the class when `jwstate` is `true`, and the os is `ios`.

Comment (pajong) [reply src/js/view/view.js]: `NATIVE_FULLSCREEN` event only fires with `event.jwstate = true` if it is iOS fullscreen.
This is the only place where `NATIVE_FULLSCREEN` is fired: (https://github.com/jwplayer/jwplayer/blob/master/src/js/providers/html5.js#L670)

Comment (robwalch) [reply src/js/view/view.js]: You think so, but it’s not guaranteed. For that reason, you should verify the iOS.

**Pull request merged**